commit.go

```diff
@@ -47,22 +48,22 @@ func (b *Builder) Commit(dest types.ImageReference, options CommitOptions) error
 	}
 	src, err := b.makeContainerImageRef(options.Compression)
 	if err != nil {
-		return err
+		return fmt.Errorf("error recomputing layer digests and building metadata: %v", err)
 	}
 	err = copy.Image(policyContext, dest, src, getCopyOptions(options.ReportWriter))
 	if err != nil {
-		return err
+		return fmt.Errorf("error copying layers and metadata: %v", err)
 	}
 	if len(options.AdditionalTags) > 0 {
 		switch dest.Transport().Name() {
 		case storage.Transport.Name():
 			img, err := storage.Transport.GetStoreImage(b.store, dest)
 			if err != nil {
-				return err
+				return fmt.Errorf("error locating just-written image: %v", err)
```

Comment:
github.com/containers/image/transports.ImageName() should include that information, adding its value to the error message.